### PR TITLE
feat: improves status panel layout

### LINKED ISSUE #1211: padding on print thumbnail making time switch between 1 and 2 lines
### Fluidd Version

1.26.0

### Browser

Chrome

### Device

Desktop PC

### Operating System

Windows

### What happened

When using a 4 column layout on my 27" 4k monitor the extra padding on the print thumbnail in the status widget means there is not enough space for the time estimates. But only just not enough, so every time the seconds changes from 1 to 2 digits it switches between 1 and 2 lines and pushes the whole column up and down and is very annoying. This padding is added by the button class. When the padding is removed there is enough room for the whole time on a single line

video of it changing as the time changes
https://github.com/fluidd-core/fluidd/assets/17092573/4f27b64e-f6d3-4a8d-aebe-90c89b5ce2e3

the extra padding because its a button
image

with the padding removed there is enough room for the time
image

### What did you expect to happen

no have padding taking up extra space. i dont think the padding on the button itself is nesaserry here because there is enough space around it from the layout of the status widget

### How to reproduce

make a column layout and change the window size so that the status widget just wide enough to show the thumbnail but not wide enough to show the full time. i get that this is pretty spesific but i dont think my particular setup would be that uncommon. it just takes a bit of messing with things to reproduce the issues without it

### Additional information

discord link: https://discord.com/channels/762863525818662974/762864050652053534/1165204358300962857

i have also had a look at fixing this by simply removing the padding and it works fine
https://github.com/TomW1605/fluidd/commit/e22472161eacb966ae72c7f4e98d51c98aa7cf0b

### LINKED ISSUE #1251: consider a differenet layout for cluttered printing status
### Fluidd Version

develop

### Browser

Chrome

### Device

Desktop PC, Laptop

### Operating System

Linux

### What happened

When printing this is what the status panel looks like.

image

I am not sure this is the best use of space along with circular progress bar

### What did you expect to happen

Consider changing the layout. Maybe we can get creative and put the thumbnail inside the circle for progress bar and then have the progress bar indicator on the outside border? or just even stack the thumbnail and progress bar.

image
 

### How to reproduce

Print in 3 panel mode. on largish monitor. 

### Additional information

None.

### PR DESCRIPTION
**Pull request by vajonam**
one possible alternate layout for the status panel.

## Before

image

## After

image

Fixes #1251
Fixes #1211 

Signed-off-by: vajonam <152501+vajonam@users.noreply.github.com>

### CODE REVIEW

File: src/components/widgets/status/StatusTab.vue [hunk 1]
```diff
@@ -18,16 +18,35 @@
...
+              height="100%"
+              class="print-thumb"
+              @click="handleViewThumbnail"
+            >
+              <img
+                class="print-thumb"
+                :src="thumbnail"
+              >
+            </v-btn>
+          </div>
```
Comment (matmen) [inline src/components/widgets/status/StatusTab.vue:46]: Enlarging the thumbnail doesn't work for me, this might need to be something like:

```suggestion
          <div
            class="progress-wrapper"
            style="cursor: pointer"
            @click="handleViewThumbnail"
          >
            <v-progress-circular
              :rotate="-90"
              :size="90"
              :width="7"
              :value="estimates.progress"
              color="primary"
              class="progress-circle"
            />
            <div class="thumbnail-background" />

            <div class="fill-height print-thumb">
              <img
                class="print-thumb"
                :src="thumbnail"
              >
            </div>
```

File: src/components/widgets/status/StatusTab.vue [hunk 2]
```diff
@@ -435,4 +435,52 @@ export default class StatusTab extends Mixins(StateMixin, FilesMixin) {
...
+  opacity: 0.8; /* Adjust opacity as needed */
+}
+
+.percentage-wrapper {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-43%,200%);
+
+}
```
Comment (matmen) [inline src/components/widgets/status/StatusTab.vue:484]: The indentation here is off - and please add a blank line before this section and remove the one after

**Review by matmen — CHANGES_REQUESTED** [review]

**Comment by matmen** [comment]
Thanks for your PR, I think the changes look nice. I've left a few comments in my review above

File: src/components/widgets/status/StatusTab.vue (hunk 2 above)
Comment (vajonam) [reply src/components/widgets/status/StatusTab.vue:484]: Not sure why the lint didn't catch that. will do.

File: src/components/widgets/status/StatusTab.vue (hunk 1 above)
Comment (vajonam) [reply src/components/widgets/status/StatusTab.vue:46]: @matmen  you can commit this to branch and I can test it out.

**Comment by vajonam** [comment]
@matmen  that pushes the percentage wrapper, not sure how to address that?

File: src/components/widgets/status/StatusTab.vue [hunk 3]
```diff
@@ -18,25 +17,41 @@
...
+              <img
+                class="print-thumb"
+                :src="thumbnail"
+              >
+            </div>
+            <div class="percentage-wrapper">
+              <span class="primary--text">{{ estimates.progress }}%</span>
+            </div>
+          </div>
         </v-col>
```
Comment (matmen) [inline src/components/widgets/status/StatusTab.vue:47]: Sorry, looks like I forgot a line in my selection
```suggestion
                :src="thumbnail"
              >
            </div>
          </div>
          <div class="percentage-wrapper">
            <span class="primary--text">{{ estimates.progress }}%</span>
          </div>
        </v-col>
```

File: src/components/widgets/status/StatusTab.vue [hunk 4]
```diff
@@ -2,9 +2,8 @@
   <div>
     <!-- Only show the linear progress for mdAndDown -->
     <v-progress-linear
-      v-if="
-        progressVisible &&
-          $vuetify.breakpoint.mdAndDown
+      v-if="progressVisible &&
+        $vuetify.breakpoint.mdAndDown
```
Comment (matmen) [inline src/components/widgets/status/StatusTab.vue:6]: LGTM functionality wise now, but I'd prefer it if unrelated lines weren't changed - looks like these were auto-formatted? Applies to https://github.com/fluidd-core/fluidd/pull/1252/files#diff-8014f06344747bd0e9f2beb41bbca3f9938e86a5228dbdb110ece4fb4acc3860R52-R139 too

Comment (vajonam) [reply src/components/widgets/status/StatusTab.vue:6]: oops was trying to fix the other .css formatting will roll it back.

**Comment by vajonam** [comment]
should be sorted now!

**Review by matmen — APPROVED** [review]
LGTM, thanks!

File: src/components/widgets/status/StatusTab.vue [hunk 5]
```diff
@@ -422,17 +420,71 @@ export default class StatusTab extends Mixins(StateMixin, FilesMixin) {
...
+  border-radius: 50%;
+  opacity: 0.8;
+  /* Adjust opacity as needed */
+}
+
+.percentage-wrapper {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-43%, 200%);
```
Comment (vajonam) [inline src/components/widgets/status/StatusTab.vue:488]: @matmen  not to happy with this absolute positioning.. probably a better way to do this. Am sure @pedrolamas  when we gets to looking into this might have some other ideas. if I find a better way to do this without using numbers I will update it

Comment (vajonam) [reply src/components/widgets/status/StatusTab.vue]: think i got it. added some rows. and center justify the content. looks clearer and less fiddly.

**Comment by vajonam** [comment]
@matmen can you test this latest commit, because it makes up much less room, I am proposing we used in mdAndUp breakpoints.

**Review by matmen — DISMISSED** [review]
LGTM!

**Comment by pedrolamas** [comment]
I've made some tweaks and simplified the styles, will perform some more tests to ensure I haven't broken anything before merging down.

**Review by pedrolamas — DISMISSED** [review]

File: src/components/widgets/status/StatusTab.vue [hunk 6]
```diff
@@ -405,4 +395,18 @@ export default class StatusTab extends Mixins(StateMixin, FilesMixin, ToolheadMi
...
+  .progress-button {
+    width: 90px !important;
+    height: 90px !important;
+    border-radius: 50%;
+    overflow: hidden;
+  }
+
+  .progress-button-image {
+    max-width: 71px;
+    max-height: 71px;
```
Comment (matmen) [inline src/components/widgets/status/StatusTab.vue:408]: With a 71px image, I've noticed the thumbnail is shifted 1px to the right and bottom, so I'd imagine this will need to be an even number:
image
image

**Review by matmen — COMMENTED** [review]
I've tested the changes and they seem fine on my end, but the positioning of the image isn't quite centered now:

File: src/components/widgets/status/StatusTab.vue (hunk 6 above)
Comment (pedrolamas) [reply src/components/widgets/status/StatusTab.vue:408]: Very nice catch!! Took me a bit to repro this as I normally use a 4K 200% scale monitor and it worked fine there, only when I moved to the 1080p 100% scale one I could see the problem!

**Comment by rursache** [comment]
please add a setting for this, i want to use the "old" layout. my thumbnails contain information in each corner which the new layout cuts off

**Comment by vajonam** [comment]
can you post a screenshot of the old layout with your thumb info? just curious to see.

**Comment by rursache** [comment]
> can you post a screenshot of the old layout with your thumb info? just curious to see.

here is an example of thumbnail elegoo cura is generating for each sliced model. if you put it inside a circle, you lose all the useful details.

knight_bust-300x300

how it looks in fluidd now:

Screenshot 2024-01-12 151353

how it looked before:

Screenshot 2024-01-12 151438 aaa

i think its obvious which one was better (at least for me and elegoo users). 

please add a setting toggle for this layout change. thanks!

**Comment by vajonam** [comment]
I disagree there is no saved space, in the cases where there is longer filename it helps with the layout. However, I understand your issue. click on the thumbnail, will bring up full thumb. I get the issue.

**Comment by pedrolamas** [comment]
@rursache I personally wasn't aware of this possibility though I can perfectly see your point.

I've discussed this with @vajonam and we are going to add an option to pick between the "classic" layout (the existing one where progress and thumbnail is separate) and the new "compact" layout (that matches the changes on this PR), that way users can choose which one is better for their purposes.

**Comment by rursache** [comment]
@pedrolamas & @vajonam thank you for your effort and consideration!

**Comment by rursache** [comment]
@pedrolamas was the option added? where can i find it and switch to the old layout? thanks

**Comment by yw4z** [comment]
296250408-71128d49-9f5d-4302-87c4-8f7d22e1d1dd-2
hi just wanted to share ideas
• moved percentage to title bar
• used progress bar at full length below title bar and this allows to using gcode preview bigger
• added exclude as button. button type is different before

**Comment by pedrolamas** [comment]
> @pedrolamas was the option added? where can i find it and switch to the old layout? thanks

Just created the PR!

**Comment by rursache** [comment]
> 296250408-71128d49-9f5d-4302-87c4-8f7d22e1d1dd-2 hi just wanted to share ideas • moved percentage to title bar • used progress bar at full length below title bar and this allows to using gcode preview bigger • added exclude as button. button type is different before

looks great!

> > @pedrolamas was the option added? where can i find it and switch to the old layout? thanks
> 
> Just created the PR!

amazing!
